### PR TITLE
Add: Backup Retention Disclaimer Information

### PR DESCRIPTION
**Pull request by CodeyGuyDylan**
#### Proposed Changes

Add a disclaimer to the bottom of Jetpack Backup Product cards that links to a FAQ on the pricing page that further describes the disclaimer.

When the Product Card is already on the pricing page, the link should jump down to the FAQ and open it up for the user.
When the Product Card is being used elsewhere, it should open a new window to https://cloud.jetpack.com/pricing and jump down to the FAQ and open it for the user.

P2 context: pbuNQi-3b3-p2

#### Testing Instructions

1. Check out this branch via `git switch add/backup-retention-disclaimer`
2. Run `yarn start` to start Calypso Blue
3. Navigate to http://calypso.localhost:3000/jetpack/connect/store and scroll until you can fully see the Jetpack backup card. There should be a disclaimer and link on the bottom
Screen Shot 2022-07-22 at 4 22 51 PM
4. Click on "Learn More." This should jump you down to the FAQ question and have it open when you get there
Screen Shot 2022-07-22 at 4 24 14 PM
5. If you open and close the FAQ several times, you should see the hash value in the URL being removed when you close it, and added when you open it. This is to make it more convenient for users to link to specific questions, as well as make it easier to open up the FAQ again if the link is clicked a second time and the user had closed the FAQ from the first time.
6. Stop the local process and run `yarn start-jetpack-cloud`. Then navigate to http://jetpack.cloud.localhost:3000/pricing and repeat steps 3-5. Everything should work exactly the same as in Jetpack Connect
7. Navigate to http://jetpack.cloud.localhost:3000/pricing/storage/{ your-site }?calypso_env=development. From here you should see the same disclaimer on both of the cards that show up
Screen Shot 2022-07-22 at 4 34 59 PM
8. Click on this link, and a new window to http://jetpack.cloud.localhost:3000/pricing/ should pop up and take you directly to that FAQ question opened up just as before. Make sure you have `?calypso_env=development`, otherwise it will just take you to the live site
9. (optional) if you know of any other places these backup product cards are used on Clayp

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/faq/index.tsx [hunk 1]
```diff
@@ -51,9 +67,10 @@ const JetpackFAQ: React.FC = () => {
 			<section className="jetpack-faq">
 				<h2 className="jetpack-faq__heading">{ translate( 'Frequently Asked Questions' ) }</h2>
 				<FoldableFAQ
-					id="priority-support"
+					id="getting-started"
```
Comment (CodeyGuyDylan) [inline client/my-sites/plans/jetpack-plans/faq/index.tsx:70]: This had the same `id` as another FAQ, I assume it was mishap so I went ahead and changed this one to one that made sense

File: packages/calypso-products/src/translations.tsx [hunk 2]
```diff
@@ -261,6 +261,32 @@ export const getJetpackProductsTaglines = (): Record<
...
+	);
+	const backupT2Disclaimer = translate(
+		'* Subject to your usage and storage limit. {{link}}Learn more{{/link}}.',
+		{
+			components: {
+				link: <a href="#backup-storage-limits-faq"></a>,
+			},
+		}
+	);
+
```
Comment (CodeyGuyDylan) [inline packages/calypso-products/src/translations.tsx:309]: Felt it best to leave T1 and T2 separate in case we ever wanted to slightly change the text for one of them. I saw similar patterns elsewhere in this file

**Comment by CodeyGuyDylan** [comment]
You'll see a couple places in this PR where I removed several instances of `React.` (for example `React.FC`) and imported them directly instead as types or functions. Hopefully it doesn't messy up the PR too much but I felt it was cleaner to not have to import all of React each time

File: packages/calypso-products/src/translations.tsx [hunk 3]
```diff
@@ -261,6 +261,40 @@ export const getJetpackProductsTaglines = (): Record<
...
+	return {
+		[ PRODUCT_JETPACK_BACKUP_T1_YEARLY ]: backupT1Disclaimer,
+		[ PRODUCT_JETPACK_BACKUP_T1_MONTHLY ]: backupT1Disclaimer,
+		[ PRODUCT_JETPACK_BACKUP_T2_YEARLY ]: backupT2Disclaimer,
+		[ PRODUCT_JETPACK_BACKUP_T2_MONTHLY ]: backupT2Disclaimer,
+	};
```
Comment (elliottprogrammer) [inline packages/calypso-products/src/translations.tsx]: I'm not completely sure if this is the code location that is causing the issue I'm experiencing, but if it is, maybe we need to add the Jetpack Security T1 & T2 products into this object?

**Anyway, here's the issue:**

With a site that owns _Jetpack Security_ (productSlug: `plan_jetpack_security_t1_yearly`), the disclaimer is not displaying on the `/pricing/storage/:siteSlug` page.

**Screenshot:**
Markup 2022-07-29 at 11 45 01

File: client/my-sites/plans/jetpack-plans/faq/index.tsx [hunk 4]
```diff
@@ -62,10 +79,21 @@ const JetpackFAQ: React.FC = () => {
...
 					) }
 				</FoldableFAQ>
+				<FoldableFAQ
+					id="backup-storage-limits"
+					question={ translate( 'How do backup storage limits work?' ) }
+					onToggle={ onFaqToggle }
+					className="jetpack-faq__section"
+				>
+					{ translate(
+						"If your site's Backup storage limit is reached, your older backups will be deleted. Depending on the size of your site and your site's Backup storage limit, your site's backup retention period may be reduced down to 7 days of your most recent backups. You will still be able to restore existing backups, but new site updates will not be backed up until you free up storage or upgrade your storage limit here"
```
Comment (elliottprogrammer) [inline client/my-sites/plans/jetpack-plans/faq/index.tsx]: In the FAQ where is says, "or upgrade your storage limit here", should the word "_here_" be a **link** to the storage upgrade page?

**Review by elliottprogrammer — COMMENTED** [review]
Overall this is looking good, however I found an issue on the `/pricing/storage/:site` page with sites owning Jetpack security (See comment below), and just a small question about the Storage Limits FAQ text. Let me know what you think. 🙂

File: packages/calypso-products/src/translations.tsx (hunk 3 above)
Comment (CodeyGuyDylan) [reply packages/calypso-products/src/translations.tsx]: Nice catch! So strange that this only happens when a site owns Jetpack Security. On the same page on a site that has Jetpack Complete, it shows up just fine. I'll take a look at this

Comment (elliottprogrammer) [reply packages/calypso-products/src/translations.tsx]: Oh! Hmm, yes that _is_ strange. 🤔 I'm not sure offhand what would be causing that. 🤔

Comment (CodeyGuyDylan) [reply packages/calypso-products/src/translations.tsx]: The problem those cards technically have the "plan" type of item instead of the "product" because it wants to upgrade the JP Security plan. So I added it to the plan builder but now it's showing up on the pricing page on the security plan 😭. Will fix this next hahaha

Comment (CodeyGuyDylan) [reply packages/calypso-products/src/translations.tsx]: Oh MAN was that a rabbithole haha, turns out in `client/my-sites/plans/jetpack-plans/storage-tier-upgrade/hooks/use-get-storage-upgrade-product.ts`, the "features" are bing overridden manually after the `client/my-sites/jetpack-plans/slug-to-selector-product.ts` function (that is used for all product cards) runs. 

Therefore, the disclaimer that I was setting in `slug-to-selector-products.ts` wasn't using the proper features to determine whether or not to display. So the only solution I could think of was to add the disclaimer in the `use-get-storage-upgrade-product` function as well so it uses the correct feature set to determine whether or not to show. 

Anyway, long story short, problem should be fixed!

File: client/my-sites/plans/jetpack-plans/faq/index.tsx (hunk 4 above)
Comment (CodeyGuyDylan) [reply client/my-sites/plans/jetpack-plans/faq/index.tsx]: Nice catch, I think the "here" wasn't present in the initial mockup but got added later on, and then I just copied and pasted. I'll link this to the https://cloud.jetpack.com/pricing/storage/:site url (Thats probably best right?)

Comment (CodeyGuyDylan) [reply client/my-sites/plans/jetpack-plans/faq/index.tsx]: Actually unless they have the `:site` in the URL, this would just send them to the Pricing page. I'll have to see if there's a better place to link them to

Comment (elliottprogrammer) [reply client/my-sites/plans/jetpack-plans/faq/index.tsx]: > I'll link this to the https://cloud.jetpack.com/pricing/storage/:site url (Thats probably best right?)

Yes, that was my first thought...  But now I just realized that in some scenarios, there is no site in context, like when visiting `cloud.jetpack.com/pricing`.  So I would say we either completely remove the word "here", Or another option might be to add the "here" link only if/when the site is in context, otherwise leave it out. So whatever you think, it's your call. 🙂

File: packages/calypso-products/src/translations.tsx (hunk 3 above)
Comment (elliottprogrammer) [reply packages/calypso-products/src/translations.tsx]: Oh wow, tricky.. Nice work figuring that out! 😀

File: client/my-sites/plans/jetpack-plans/faq/index.tsx (hunk 4 above)
Comment (CodeyGuyDylan) [reply client/my-sites/plans/jetpack-plans/faq/index.tsx]: I consulted the team and we decided to remove it :)

**Review by elliottprogrammer — APPROVED** [review]
Awesome! Nice work debugging the issue, It's working all good now!
LGTM! 👍

**Comment by CodeyGuyDylan** [comment]
Sorry everybody who just got a review ping, I made a mistake when rebasing (as you can probably tell)

**Comment by CodeyGuyDylan** [comment]
rebased branch to attempt to get E2E tests to pass. Previously there was a test failing that had nothing to do with the changes

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7462122

Thank you @CodeyGuyDylan for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7462122).
